test/e2e/basic/netdev_set_channels.go

```diff
@@ -0,0 +1,163 @@
+		ginkgo.AfterEach(func() {
+			ginkgo.By("cluster changes rollback")
+			if node != nil {
+				util.ExecAndLogCommand("oc", "label", "node", "--overwrite", node.Name, nodeLabelNetdev+"-")
+			}
+			util.ExecAndLogCommand("oc", "delete", "-n", ntoconfig.OperatorNamespace(), "-f", profileNetdev)
+		})
+
+		ginkgo.It("adjust netdev queue count for physical network devices via ethtool", func() {
+			var (
+				phyDev  string
+				explain string
+			)
+			const (
+				pollInterval = 5 * time.Second
+				waitDuration = 5 * time.Minute
+				chLen        = 2 // the number of multi-purpose channels to set by profileNetdev
+			)
+			cmdGetPhysicalDevices := []string{"find", "/sys/class/net", "-type", "l", "-not", "-lname", "*virtual*", "-printf", "%f "}
+
+			ginkgo.By("getting a list of worker nodes")
+			nodes, err := util.GetNodesByRole(cs, "worker")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(len(nodes)).NotTo(gomega.BeZero(), "number of worker nodes is 0")
+
+			node = &nodes[0]
+			ginkgo.By(fmt.Sprintf("getting a tuned Pod running on node %s", node.Name))
+			pod, err := util.GetTunedForNode(cs, node)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			util.Logf("found Pod %s running on node %s", pod.Name, node.Name)
+
+			ginkgo.By(fmt.Sprintf("getting a list of physical network devices: %v", cmdGetPhysicalDevices))
+			phyDevs, err := util.ExecCmdInPod(pod, cmdGetPhysicalDevices...)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			for _, d := range strings.Split(phyDevs, " ") {
+				if d == "" {
+					continue
+				}
+				// See if the device 'd' supports querying the channels.
+				_, err := util.ExecCmdInPod(pod, "ethtool", "-l", d)
+				if err == nil {
+					// Found a device that supports querying the channels
+					phyDev = d
+					break
+				}
+			}
+
+			if phyDev == "" {
+				util.Logf(fmt.Sprintf("no network devices supporting querying channels found on node %s; skipping test", node.Name))
+				return
+			}
+
+			cmdCombinedChannelsCurrent := []string{"bash", "-c", "ethtool -l " + phyDev + "| sed -n '/Current hardware settings/,/Combined:/{s/^Combined:\\s*//p}'"}
+			cmdCombinedChannelsMax := []string{"bash", "-c", "ethtool -l " + phyDev + "| sed -n '/Pre-set maximums/,/Combined:/{s/^Combined:\\s*//p}'"}
+
+			ginkgo.By(fmt.Sprintf("using physical network device %s for testing", phyDev))
+			out, err := util.ExecCmdInPod(pod, cmdCombinedChannelsMax...)
+			channelMaximumCombined, err := strconv.Atoi(strings.TrimSpace(out))
+			if err != nil || channelMaximumCombined <= 1 {
+				util.Logf(fmt.Sprintf("network device %s does not seem to support querying channels [%v] on node %s; skipping test: %v",
+					phyDev, channelMaximumCombined, node.Name, err))
+				return
+			}
+
+			out, err = util.ExecCmdInPod(pod, cmdCombinedChannelsCurrent...)
+			channelOrigCombined, err := strconv.Atoi(strings.TrimSpace(out))
```

Comment:
if `channelOrigCombined == chLen` the should be skipped I guess

Reply:
I don't think it is needed.  I'm lowering the channel to "1" down below. What is missing though is a check it was really lowered to 1, otherwise one might get a false positive.  Will fix, thanks!

Reply:
indeed, makes sense:
1. get the original value
2. manually set the current value to 1
3. use NTO to set the value to 2
4. wait for the value to become 2
5. set the original value back

and what's missing is 2.5/ wait for the value to become 1

Reply:
Correct and fixed (2.5) by the latest commit I hope.  You don't really need to "wait" (as in poll) though.  At least I hope these settings (by calling `ethtood` manually) take effect immediately.  If not, the test will fail.